lib/maths/CTimeSeriesChangeDetector.cc

```diff
@@ -178,40 +195,36 @@ double CUnivariateTimeSeriesChangeDetector::decisionFunction(std::size_t& change
                        noChangeBic - candidates[1].first};
     double expectedEvidence{noChangeBic - (*candidates[0].second)->expectedBic()};
 
-    double x[]{evidences[0] / m_MinimumDeltaBicToDetect,
-               2.0 * (evidences[0] - evidences[1]) / m_MinimumDeltaBicToDetect,
-               evidences[0] / EXPECTED_EVIDENCE_THRESHOLD_MULTIPLIER / expectedEvidence,
-               static_cast<double>(m_TimeRange.range() - m_MinimumTimeToDetect) /
-                   static_cast<double>(m_MaximumTimeToDetect - m_MinimumTimeToDetect)};
+    double x[]{
+        evidences[0] / m_MinimumDeltaBicToDetect,
+        2.0 * (evidences[0] - evidences[1]) / m_MinimumDeltaBicToDetect,
+        evidences[0] / EXPECTED_EVIDENCE_THRESHOLD_MULTIPLIER / expectedEvidence,
+        std::max(static_cast<double>(m_TimeRange.range() - m_MinimumTimeToDetect), 0.0) /
+            static_cast<double>(m_MaximumTimeToDetect - m_MinimumTimeToDetect)};
     double p{CTools::logisticFunction(x[0], 0.05, 1.0) *
              CTools::logisticFunction(x[1], 0.1, 1.0) *
              (x[2] < 0.0 ? 1.0 : CTools::logisticFunction(x[2], 0.2, 1.0)) *
              CTools::logisticFunction(x[3], 0.2, 0.5)};
-    LOG_TRACE("p(" << (*candidates[0].second)->change()->print() << ") = " << p
-                   << " | x = " << core::CContainerPrinter::print(x));
+    LOG_TRACE(<< "df(" << (*candidates[0].second)->change()->print() << ") = " << p / 0.03125
```

Comment:
It would make it clearer to the reader to use `MAXIMUM_DECISION_FUNCTION * p` rather than `p / 0.03125` even though they evaluate to the same thing.

Reply:
agreed